contracts/libraries/TokenConservation.sol

```diff
@@ -62,12 +62,15 @@ library TokenConservation {
       */
     function findPriceIndex(uint16 tokenId, uint16[] memory tokenIdsForPrice) private pure returns (uint256) {
         // binary search for the other tokens
+        if (tokenId == 0) {
```

Comment:
I think duck-taping this one-off logic here is somewhat hacky. Can we adjust the comments to document why this works and what this function is really doing now?

Reply:
Great idea! There are potentially a few other places where the comments should be updated. I don’t necessarily think this is duct tape, since this is exactly how the logic is intended to work. However, I can try to think of some super clever, even more natural implementation of this logic if need be...